### ISSUE TITLE
croutonpowerd: Using it properly?

### ISSUE THREAD

**Issue by darethehair** [dnschneid/crouton#3327]
I have installed a webserver into my Debian 'jessie' crouton install on a Dell 11 Chromebook.  This works great!  However, with the lid closed, in a matter of hours the wifi connection shuts down (?) and it ceases to function as intended i.e. a webserver.  Simply opening the lid is sufficient to make it work again.  

I thought I had done what I needed by adding this line to '/etc/rc.local':

`/usr/local/bin/croutonpowerd -i &`

Am I doing it wrong?  Thanks for any help!

**Comment by dnschneid** [comment]
croutonpowerd will inhibit the system from going to sleep due to idleness, but not due to lid close events.

You can stop powerd entirely (`sudo stop powerd`) which will stop the system from responding to lid events (and also remove the need for croutonpowerd), but depending on your system, the screen/backlight may remain on when you close the lid, which will heat up the device a bit and potentially cause it to throttle.

I actually am using an Acer Chromebook 13 (the Tegra one) as a server, and use the following script to ensure the screen turns off when the lid is closed:

```
#!/bin/sh -e
stop powerd || true
stdbuf -oL hexdump /dev/input/by-path/platform-gpio-keys.*-event | \
    while read _ _ _ _ _ btn zeroa closed zerob; do
        if [ ! "$btn" = 0005 -o ! "$zeroa" = 0000 -o ! "$zerob" = 0000 ]; then
            continue
        fi
        if [ "$closed" = 0000 ]; then
            brightness=200
        else
            brightness=0
        fi
        for f in /sys/class/backlight/*/brightness; do
            echo "$brightness" > "$f"
        done
    done
```

This literally monitors the gpio input device to see if the lid switch is triggered, and turns the backlight on and off accordingly.  You need to run it as root.  You may also need to change the paths or button numbers to match whatever your device internally uses.

Nowadays, a better solution would be to trick powerd into being in dock mode, which it enters if you have an external display attached.  Dock mode + croutonpowerd would be way cleaner than stop-powerd+hacky-script, but I don't know if there's a way to force dock mode.

**Comment by dnschneid** [comment]
Actually, it looks like we may be able to use dbus_send to call the powerd method SetIsProjecting with a value of true, which might stick until you reconfigure displays.  If that works, this may be worth adding to croutonpowerd (and certainly better than hacky scripts)...

**Comment by dnschneid** [comment]
Try running this: 

```
host-dbus dbus-send --system --dest=org.chromium.PowerManager --type=method_call /org/chromium/PowerManager org.chromium.PowerManager.SetIsProjecting boolean:true
```

**Comment by darethehair** [comment]
Thanks for assisting with this challenge!  I have issued that command -- without any feedback or errors -- while signed on as a normal user inside my chroot.  Will there be any way to find out if it is working, other than noticing that my system doesn't eventually fall asleep after many hours?  Is this something that should be done in addition to -- or instead of -- the 'croutonpowerd' command?

**Comment by dnschneid** [comment]
you should do that in addition to croutonpowerd.  Then close the lid and see if your server is still accessible, I guess?

**Comment by darethehair** [comment]
Oh, I did this while the lid was already closed (as it normally is) via ssh'ing into the system (while it is normally accessible).  And yes, the system and webserver remains accessible even after doing this.

**Comment by dnschneid** [comment]
Oh wait, I misread your original message.  How on earth did that ever work?  Anyway, wait and see, I suppose?

**Comment by darethehair** [comment]
Yes, to be clear, closing the lid doesn't put my system to sleep -- that only occurs many hours later (12?  24?), so I speculated that 'croutonpowerd' is doing *something* good for me, but that even it wasn't designed (?) to act for so many hours.  Of course I have no real idea of what I am talking about :)

**Comment by darethehair** [comment]
Footnote: I recently purchased the Dell 11 Chromebook 'as is', with a 'damaged' screen that would only display half of what it should.  This did not matter to me, since my intent was to physically *remove* the display panel and turn the Chromebook into a sortof 'Chromebox' by doing this.  I assumed that the device would act like the screen was perpetually 'open'.  However, when I disassembled it, I tried re-seating the display, and returned it almost entirely back to working operation.  I no longer had a desire to rip the screen off :)  And here I am struggling with the 'normal' behavior of a Chromebook to keep it from falling asleep because the display is open :)
